### DISCUSSION TITLE
@next/swc-win32-x64-msvc

### DISCUSSION THREAD

**Discussion by diphan-source** [vercel/next.js#31073] [Help] [answered]
I'm getting this error when I run `yarn dev`
I'm using 
`win 11 64 bit`
`node v16.13.0`
`yarn v1.22.17`
`next v12.0.3`

here are the error logs
```bash
diphan@diphan MINGW64 ~/outbox (ft-login-EM-1)
$ yarn dev
yarn run v1.22.17
$ next dev
ready - started server on 0.0.0.0:3000, url: http://localhost:3000
Error: The specified module could not be found.
\\?\C:\Users\diphan\outbox\node_modules\next\node_modules\@next\swc-win32-x64-msvc\next-swc.win32-x64-msvc.node
    at Object.Module._extensions..node (node:internal/modules/cjs/loader:1183:18)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (C:\Users\diphan\outbox\node_modules\next\dist\build\swc\index.js:29:20)
    at Module._compile (node:internal/modules/cjs/loader:1101:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12) {
  code: 'ERR_DLOPEN_FAILED'
}
error - Failed to load SWC binary, see more info here: https://nextjs.org/docs/messages/failed-loading-swc
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Any help regarding this error will be appreciated. Thank you

**Comment by femostic4j** [comment]
What you could do to solve that issue is to re-install the latest version of node

**Comment by dpkreativ** [comment, marked as answer]
Hi! Try downloading Microsoft Visual C++ Redistributable.
I had the same issue and this fixed it for me.
Get it here: https://aka.ms/vs/17/release/vc_redist.x64.exe

**Reply by KartikCD** [reply]
This solution helped me!!! Thankss

**Reply by kiranpoornavishnuM** [reply]
Thanks! This one helped

**Reply by dhananjayhegde** [reply]
I reset my Windows and after that I started getting something similar error:

"@next\swc-win32-x64-msvc is not a valid Win32 application".  

Struggled for about half an hour and finally this worked like a charm, thank you so much!!!

**Reply by asifali-7890** [reply]
> Hi! Try downloading Microsoft Visual C++ Redistributable. I had the same issue and this fixed it for me. Get it here: https://aka.ms/vs/17/release/vc_redist.x64.exe

BELIEVE ME, IT WORKS. JUST INSTALLED THIS THEN RUN, YOU WILL SEE IT WORKS.

**Reply by hoangvu12** [reply]
This should be mentioned at https://nextjs.org/docs/messages/failed-loading-swc

**Reply by realtouseef** [reply]
Thank you, it worked for me too.

**Reply by megh7199** [reply]
This worked finally after all alternatives failed.

**Reply by obviouslyquell** [reply]
> I reset my Windows and after that I started getting something similar error:
> 
> "@next\swc-win32-x64-msvc is not a valid Win32 application".
> 
> Struggled for about half an hour and finally this worked like a charm, thank you so much!!!

Thank u mate

**Reply by Salmanghouri** [reply]
This Link Work for me . I stuck 2 hour .. Thanks for this... https://aka.ms/vs/17/release/vc_redist.x64.exe

**Reply by zeeshan1791** [reply]
thank you its work

**Reply by ChaniceIrungu** [reply]
Great, this also works for me.
Thank you.

**Reply by unixapple** [reply]
> This Link Work for me . I stuck 2 hour .. Thanks for this... https://aka.ms/vs/17/release/vc_redist.x64.exe

also stuck with tauri project build on a new windows 10 guest, that inspired and helped a lot...

**Reply by MMPgenave** [reply]
I performed npm update and also getting latest version of Microsoft Visual C++ Redistributable, but still having this issue.
please help, stuck for 2 days!.

**Reply by MMPgenave** [reply]
I just solve this issue by installing @next/swc-win32-x64-msvc on node.

`npm i @next/swc-win32-x64-msvc`

**Comment by paulwongx** [comment]
I have a similar issue but it says
```js
error An unexpected error occurred: "EPERM: operation not permitted, unlink 
'E:\\Dev\\my_project\\node_modules\\@next\\swc-win32-x64-msvc\\next-
swc.win32-x64-msvc.node'".
```
Is it the same issue?

**Reply by dpkreativ** [reply]
Hi! Have you sorted this issue out yet?

**Reply by Aurase-xnx** [reply]
This happens when your app is still running, just stop your app and it should work

**Comment by Rohullah-Sarabi** [comment]
I installed  Microsoft Visual C++ Redistributabl , but again show the same error. what should I do?

**Reply by tochukwujonah** [reply]
I'm experiencing the same issue, please have you found a fix?

**Reply by Ayothegod** [reply]
hi, I'm experiencing this. have you been able to fix it?

**Reply by tuskermanshu** [reply]
I'm in the same situation, installed the program but run it with the same error

**Reply by tuskermanshu** [reply]
Deleting node_modules and reinstalling it solves the problem.

**Reply by bigbigbigwaterbucket** [reply]
thank you it works

**Reply by techvishnu** [reply]
what did u to make it work, im still facing the problem

**Reply by virajithdissanayaka** [reply]
I'm also still facing this issue. How can I fix that?

**Reply by bigbigbigwaterbucket** [reply]
try deleting node_modules and reinstalling it

**Reply by virajithdissanayaka** [reply]
I deleted and reinstalled the node _modules multiple times. But still facing the issue

**Reply by techvishnu** [reply]
me too, i think i would have to start using linux again windows is bad

**Reply by Hamza3357** [reply]
i am still facing the same error. have you found any solution??\

**Reply by pinological** [reply]
It's 2024, and I've just started learning Next.js. I used npx create-next-app@latest, and when I try to run npm run dev, I'm encountering the same error.

**Comment by MDileg** [comment]
I installed linux kali and windows 10 but Shows same isues..

Attempted to load @next/swc-win32-x64-msvc, but an error occurred: \\?\C:\Users\Akgurban\Desktop\peykam-next-app\node_modules\@next\swc-win32-x64-msvc\next-swc.win32-x64-msvc.node is not a valid Win32 application.

**Comment by insekia-Dev** [comment]
I had a similar issue. Firstly start by installing the latest version `npm i @next/swc-win32-x64-msvc`
After run again `npm run build`
After clear the cache by running `npx next clear`
Now run `npm run dev`
The issue should be fixed.

**Reply by xchaoscn** [reply]
helpful answer

**Reply by sYsguard** [reply]
Thanks, it is solved the issue

**Comment by clicktodev** [comment]
deleting the `node_modules `and` .next` folders worked for me

**Comment by CaptainN** [comment]
When I used to develop on Windows (and I was very very sorry I did), I used to develop all node.js based apps within WSL2, from within the simulated home directory. node.js runs about twice as fast there (ore more - I'm talking 30 minutes builds down to 10 minutes for some complex apps). You can also open that in a native feeling way in VS Code, which is my editor of choice. Just a random bit of advice no one asked for.
